### ISSUE TITLE
Resource field is empty when importing data

### ISSUE THREAD

**Issue by nortigo** [django-import-export/django-import-export#1831]
Hi,

I have Django 4.2.13 with django-import-export 4.0.2. 

I have wrote an admin class that inherits from `ImportExportModelAdmin` and defined the `resource_classes` property.

But for some reason, the field `resource` is empty when trying to import. And the fields are not displayed in `This importer will import the following fields:`.

Here is a screenshot:
![image](https://github.com/django-import-export/django-import-export/assets/20732921/5045ace1-014d-4cc0-9997-eb4a837ca983)

What could be reason? What am I missing?

Thank you

**Comment by matthewhegarty** [comment]
Can you give any more information to go on

- Does it work in v4.0.1 or v3.3.8?
- You can enable [logging at debug level](https://django-import-export.readthedocs.io/en/latest/testing.html#enable-logging) to see if anything is raised
- Can you step through with a debugger to identify the issue (this will be the quickest and best way to resolve)
- Can you reproduce in the [example app](https://django-import-export.readthedocs.io/en/latest/installation.html#example-app)?

**Comment by nortigo** [comment]
Hi @matthewhegarty,

Thank you for your help.

I forgot to explain bit of context. I apologize.

We have an existing project which was working fine with `Django==3.1.14` and `django-import-export==2.7.1`. 

We decided to upgrade python to 3.11.2 and then `Django==4.2.13`. I had to make some adjustment with our project since I also updated django-import-export. Then, I cannot see the resource(s) in the dropdown.

> Does it work in v4.0.1 or v3.3.8?

If was working with 2.7.1 but not since the upgrade. The problem remained with 4.0.1.

> You can enable [logging at debug level](https://django-import-export.readthedocs.io/en/latest/testing.html#enable-logging) to see if anything is raised

Logging is already enabled. No error so far.

> Can you step through with a debugger to identify the issue (this will be the quickest and best way to resolve)

I have checked `resources` param in `ImportForm.__init__` and it does contain the resource. I will investigate further to see where the information is lost.

> Can you reproduce in the example app?

I tried but I have this error:  `LookupError: setuptools-scm was unable to detect version`
